CONTRIBUTING.md

```diff
@@ -4,64 +4,98 @@ Thank you for taking the time to contribute to our organization and this reposit
+- Before opening a pull request read the [development section](#development)
+- Open a [Pull Request](#pull-requests) for others to review the branch.
 
 ## Pull Requests
+
 ---
 
-To accommodate review process we suggest that PRs are categorically broken up. Ideally each PR addresses only a single issue. Additionally, as much as possible code refactoring and cleanup should be submitted as a separate PRs from bugfixes/feature-additions. 
+To accommodate review process we suggest that PRs are categorically broken up. Ideally each PR addresses only a single issue. Additionally, as much as possible code refactoring and cleanup should be submitted as a separate PRs from bugfixes/feature-additions.
 
-If you are working on a issue and plan on contributing, please submit your PR early and make sure its opened as a `Draft`, even if it's incomplete, this indicates to the community you're working on something and allows them to provide comments early in the development process. When the code is complete it should be marked using Github's `Mark Ready` feature. This will let the maintainers know that there is a open PR for review. 
+If you are working on a issue and plan on contributing, please submit your PR early and make sure its opened as a `Draft`, even if it's incomplete, this indicates to the community you're working on something and allows them to provide comments early in the development process. When the code is complete it should be marked using Github's `Mark Ready` feature. This will let the maintainers know that there is a open PR for review.
 
 ## Development
+
 ---
 
-The code should follow [Rust Style Guide](https://github.com/rust-lang/rfcs/tree/master/style-guide). Much of the code style is captured by `rustfmt`. 
+The code should follow [Rust Style Guide](https://github.com/rust-lang/rfcs/tree/master/style-guide). Much of the code style is captured by `rustfmt`.
 
 Before opening a Pull Request please run the checks below:
 
-* Install rustfmt: `rustup component add rustfmt`
-* Install clippy: `rustup component add clippy`
+- Install rustfmt: `rustup component add rustfmt`
+- Install clippy: `rustup component add clippy`
 
-### Testing 
+### Testing
 
-Run the test suite with 
+Run the test suite with
 
-```cargo test --all-features```
+`cargo test --all-features`
 
 ### Format checking (rustfmt)
 
 Make sure your code is well-formatted by running
 
-```cargo fmt```
+`cargo fmt`
 
 ### Lint (clippy)
 
 Lint your code (i.e. check it for common issues) with:
 
-```cargo clippy```
+`cargo clippy`
+
+## Changelog
+
+Every fix, improvement, feature, or breaking change should be made in a
+pull-request that includes an update to the `CHANGELOG.md` file.
+
+Changelog entries should be formatted as follows:
+
+```
+- [module] \#xxx Some description about the change (@contributor)
+```
+
+Here, `module` is the part of the code that changed (typically a
+top-level Go package), `xxx` is the pull-request number, and `contributor`
```

Comment:
there's only a single library / module (besides examples), so this entry convention may not be worth it